src/kernel/src/main.rs

```diff
@@ -535,3 +473,89 @@ impl Default for ExecutionEngine {
         ExecutionEngine::Llvm
     }
 }
+
+#[derive(Debug, Error)]
+enum DiscordPresenceError {
+    #[error("Failed to create Discord IPC")]
+    FailedToCreateIpc(#[source] Box<dyn Error>),
+
+    #[error("Failed to update Discord presence")]
+    FailedToUpdatePresence(#[source] Box<dyn Error>),
+}
+
+#[derive(Debug, Error)]
+enum KernelError {
+    #[error("Failed to open .kernel-debug")]
+    FailedToOpenDebug(#[source] std::io::Error),
+
+    #[error("Failed to parse .kernel-debug")]
+    FailedToParseDebug(#[from] serde_yaml::Error),
```

Comment:
Better to not use `#[from]` here because `serde_yaml::Error` is a generic one, which mean if we have another code that load YAML we might accidentally using `?` and get this variant, which is not a correct one.